### PR TITLE
fix(files): failsafe when executing actions methods

### PR DESCRIPTION
**Pull request by skjnldsv**
Should avoid breaking the files app entirely when an external app have written invalid code.

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]
Good idea 👍

**Review by blizzz — APPROVED** [review]

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 1]
```diff
@@ -236,13 +243,18 @@ export default defineComponent({
...
+				if ((this.gridMode || (this.filesListWidth < 768 && action.inline)) && typeof action.title === 'function') {
+					// if an inline action is rendered in the menu for
+					// lack of space we use the title first if defined
+					const title = action.title([this.source], this.currentView)
+					if (title) return title
+				}
+				return action.displayName([this.source], this.currentView)
+			} catch (e) {
+				logger.error('Error while getting action display name', { action, e })
+				return ''
```
Comment (susnux) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:256]: There should be at least some name,  the empty string is not a good state also for ux.
Maybe use the ID.

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 2]
```diff
@@ -236,13 +243,18 @@ export default defineComponent({
...
-				if (title) return title
+			try {
+				if ((this.gridMode || (this.filesListWidth < 768 && action.inline)) && typeof action.title === 'function') {
+					// if an inline action is rendered in the menu for
+					// lack of space we use the title first if defined
+					const title = action.title([this.source], this.currentView)
+					if (title) return title
+				}
+				return action.displayName([this.source], this.currentView)
+			} catch (e) {
```
Comment (susnux) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:254]:
```suggestion
			} catch (error) {
```

So the key in the logger context is meaningful

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 3]
```diff
@@ -236,13 +243,18 @@ export default defineComponent({
...
+				logger.error('Error while getting action display name', { action, e })
```
Comment (susnux) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:255]:
```suggestion
				logger.error('Error while getting action display name', { action, error })
```

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 4]
```diff
@@ -163,7 +163,14 @@ export default defineComponent({
 			if (this.filesListWidth < 768 || this.gridMode) {
 				return []
 			}
-			return this.enabledFileActions.filter(action => action?.inline?.(this.source, this.currentView))
+			return this.enabledFileActions.filter(action => {
+				try {
+					return action?.inline?.(this.source, this.currentView)
+				} catch (e) {
+					logger.error('Error while checking if action is inline', { action, e })
```
Comment (susnux) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:170]:
```suggestion
					logger.error('Error while checking if action is inline', { action, error })
```

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 5]
```diff
@@ -257,7 +269,13 @@ export default defineComponent({
 				return
 			}
 
-			const displayName = action.displayName([this.source], this.currentView)
+			let displayName = action.id
+			try {
+				displayName = action.displayName([this.source], this.currentView)
+			} catch (e) {
```
Comment (susnux) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:275]:
```suggestion
			} catch (error) {
```

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 6]
```diff
@@ -257,7 +269,13 @@ export default defineComponent({
...
+				logger.error('Error while getting action display name', { action, e })
```
Comment (susnux) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:276]:
```suggestion
				logger.error('Error while getting action display name', { action, error })
```

**Review by susnux — DISMISSED** [review]

**Pull request merged**